router.c

```diff
@@ -1938,6 +1948,37 @@ router_rewrite_metric(
 	return 0;  /* we couldn't copy everything */
 }
 
+static inline unsigned short
+fnv1a_get_pos(
+		enum clusttype type,
+		unsigned short pos_count,
+		const char *metric,
+		const char *firstspace)
+{
+	if (type == FNV1A_H) {
+		return fnv1a_hash32(metric, firstspace) % pos_count;
+	} else {
+		/* FNV1A_SH
+		 *
+		 * The modulo operation (%), has different result in C and Python (
+		 * Python's Integer Division Floors by Guido van Rossum).
+		 *
+		 * We use the following formula for 'x % y':
+		 *     x % y =  x - y * floor(x/y), for y != 0
+		 * The above formula is copyed from Concreate Mathmatics (formula 3.21),
+		 * with a slight modification. It's compatible with Python's implementation.
+		 *
+		 * The reason we use FNV1A_SH is that our internal Python implementation
+		 * of `fvn1a.get_hash_bugfree` is a signed integer, and I have used it in
+		 * original carbon-relay. If you don't have this problem, just use FNV1A_H.
+		 */
```

Comment:
@windreamer, Guido has written a [blog](http://python-history.blogspot.com/2010/08/why-pythons-integer-division-floors.html) about the integer division operation (//) and its sibling, the modulo operation (%).

The following is pointed out by Tim pointed in the comments:

```
Should note that C Classic didn't define the sign of the result, but did require that

a == (a/b)*b + a%b

hold whenever a/b was representable. Not one C programmer in a million knew that, though ;-) 
C99 finally did insist on giving the wrong answer, seemingly just to be compatible with FORTRAN. 
```